reconcile/utils/gql.py

```diff
@@ -78,24 +85,19 @@ def __init__(self, path, msg):
 
 
 class GqlApi:
-    _valid_schemas = None
+    _valid_schemas: list[str] = []
     _queried_schemas: Set[Any] = set()
 
-    def __init__(self, url, token=None, int_name=None, validate_schemas=False):
-        self.url = url
-        self.token = token
+    def __init__(self, client: Client, int_name=None, validate_schemas=False) -> None:
         self.integration = int_name
         self.validate_schemas = validate_schemas
-        self.client = GraphQLClient(self.url)
+        self.client = client
```

Comment:
While I like the direction that you were heading in with this change, maybe passing in the `Client` and relying on a single session breaks too many assumptions with this class? This is compared to initiating a new request on each `query` like [we were doing before](https://github.com/prisma-labs/python-graphql-client/blob/master/graphqlclient/client.py#L26)?

This may or may not be the right solution, but it would maintain the functionality of the old client without needing to worry about a new set of issues (lock contention).

Reply:
Per our discussion, I removed the locks and instead went with reinitializing `Client` every time for new query.